Documentation/src/for-each.md

```diff
@@ -0,0 +1,16 @@
+# `IEnumerable` extension method `.ForEach`
+With the `.ForEach` extension method, you can invoke an action for each item in an enumerable, just like a `for each` statement would allow you to do.
+This method is already available in .NET, but just on `List`s, and it makes sense for it to be available on every enumerable.
+

```

Comment:
Here is a passage missing that ForEach is imperative and only (`Action<T>`), and should not be used to change state of anything outside the loop, and maybe give Aggregate as an functional alternative.